### ISSUE TITLE
List entries for file & namespace enums

### ISSUE THREAD

**Issue by olafmandel** [doxygen/doxygen#8274]
**Describe the bug**
Any `\bug`, `\deprecated`, `\test`, `\todo` or `\xrefitem` entry from a file-scope or namespace-scope enum is missing from their respective list under "Related pages". The paragraph in the enum documentation is generated, though. There is no warning on the output for this.

**Expected behavior**
Get an entry in the list for every of the above mentioned commands in addition to the already generated paragraph in the enum documentation.

**To Reproduce**
Self-contained example (works with default generated configuration):

```cpp
// Tested with doxygen 1.8.13, 1.8.20 & 1.9.0+git.9bf76580 and default Doxyfile

enum EnumA { };
enum class EnumB { };

namespace Namespace {
    enum EnumC { };
    enum class EnumD { };
} // Namespace

/** \file
 *  \brief Some file
 *  \bug All entries from file scope and namespace scope enums are missing
 *       in their respective lists
 *  \deprecated Dummy entry to have a deprecated list at all
 *  \test Dummy entry to have a test list at all
 *  \todo Dummy entry to have a todo list at all
 *  \xrefitem foo "Foo" "Foo list" Dummy entry to have a "Foo" list at all
 *
 *  \enum EnumA
 *  \brief Some unscoped enum in file scope
 *  \bug This entry is missing from bug list
 *  \deprecated This entry is missing from deprecated list
 *  \test This entry is missing from test list
 *  \todo This entry is missing from todo list
 *  \xrefitem foo "Foo" "Foo list" This entry is missing from "Foo" list
 *
 *  \enum EnumB
 *  \brief Some scoped enum in file scope
 *  \bug This entry is missing from bug list
 *  \deprecated This entry is missing from deprecated list
 *  \test This entry is missing from test list
 *  \todo This entry is missing from todo list
 *  \xrefitem foo "Foo" "Foo list" This entry is missing from "Foo" list
 *
 *  \namespace Namespace
 *  \brief Some namespace
 *
 *  \enum Namespace::EnumC
 *  \brief Some unscoped enum in a namespace
 *  \bug This entry is missing from bug list
 *  \deprecated This entry is missing from deprecated list
 *  \test This entry is missing from test list
 *  \todo This entry is missing from todo list
 *  \xrefitem foo "Foo" "Foo list" This entry is missing from "Foo" list
 *
 *  \enum Namespace::EnumD
 *  \brief Some scoped enum in a namespace
 *  \bug This entry is missing from bug list
 *  \deprecated This entry is missing from deprecated list
 *  \test This entry is missing from test list
 *  \todo This entry is missing from todo list
 *  \xrefitem foo "Foo" "Foo list" This entry is missing from "Foo" list
 */
```

**Version**
Observed in 

* `1.8.13` as distributed in `Debian GNU/Linux 10 (buster)`, executed under Linux 64bit
*  `1.9.0 (9bf7658013e49dabaafb818a639b5ca62f845538)` compiled from git in `Debian GNU/Linux 10 (buster)`, executed under Linux 64bit
* `1.8.20 (6e45857ba5a569049deb5788a3ed7e5109286acb*)` as distributed in the MSYS2 package `mingw-w64-x86_64-doxygen`, executed under WIn10 64bit

**Additional context**
This was originally reparted as a side-note in #6471, broken out as a separate ticket now that the main problem in the other ticket has been fixed.

**Comment by doxygen** [comment]
@olafmandel Please check if the latest commit fixes this problem.

**Comment by pinchartl** [comment]
Commit eb8b6f40d863a8d7d2798a95cbde4200e27d4945 unfortunately broke the following test case.

```cpp
/**
 * \file main.h
 * \brief Main file
 */

namespace ns {

/**
 * \namespace ns
 * \brief ns
 */

class Class1
{
public:
	enum Enum {
		Class1,
	};
};

class Class2
{
public:
	enum Enum {
		Class2,
	};
};

/**
 * \class Class1
 * \brief Class1
 */

/**
 * \enum Class1::Enum
 * \brief Class1::Enum
 *
 * \var Class1::Class1
 * \brief Class1::Enum::Class1
 */

/**
 * \class Class2
 * \brief Class2
 */

/**
 * \enum Class2::Enum
 * \brief Class2::Enum
 *
 * \var Class2::Class2
 * \brief Class2::Enum::Class2
 */

} /* namespace ns */
```

The following warning is generated.

```
/home/laurent/src/test/doxygen/enums/main.h:24: warning: Member Enum (enumeration) of class ns::Class2 is not documented.
```

**Comment by doxygen** [comment]
@pinchartl The matching rules were not strict enough. Please try again with the latest commit.

**Comment by albert-github** [comment]
At the moment the representation on the pages html/classes.html is different from the 1.8.20 version, not sure if this is from these fixes or from an earlier fix.

**Comment by pinchartl** [comment]
The regression is gone. Thank you for the quick fix !

**Comment by albert-github** [comment]
Regarding the comment: https://github.com/doxygen/doxygen/issues/8274#issuecomment-750871331 (representation on classes page) this looks like an intended change from:
```
Commit: 59372719ef31d7a76e2cfb64ab4628a315ccaf3a [5937271]
Commit Date: Monday, December 21, 2020 1:54:18 PM
Refactoring: modernize class index
```

Looks a bit more modern.

**Comment by olafmandel** [comment]
> @olafmandel Please check if the latest commit fixes this problem.

This works for me now: Thank you!

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
